base/socket.jl

```diff
@@ -2,6 +2,7 @@
 abstract IPAddr
 
 Base.isless{T<:IPAddr}(a::T, b::T) = isless(a.host, b.host)
+Base.convert{T<:Integer}(dt::Type{T}, ip::IPAddr) = dt(ip.host)
```

Comment:
`dt` and `T` are exactly equivalent and refer to the `DataType` being passed as the first argument. Like Stefan said, it's more typical to write it like he showed since you don't have to give a variable name to the first arg and the use of `T` everywhere is more obviously referring to the original type parameter.